srai/joiners/intersection_joiner.py

```diff
@@ -0,0 +1,58 @@
+"""
+Intersection Joiner.
+
+This module contains intersection joiner implementation.
+
+"""
+
+import geopandas as gpd
+import pandas as pd
+
+from srai.joiners import JoinerResult
+
+
+class IntersectionJoiner:
+    """
+    Intersection Joiner.
+
+    Intersection Joiner allows to join two GeoDataFrames and find all
+    overlapping geometries. It does not apply any grouping or
+    aggregation.
+
+    """
+
+    def join(self, regions: gpd.GeoDataFrame, features: gpd.GeoDataFrame) -> JoinerResult:
+        """
+        Join features to regions.
+
+        Args:
+            regions (gpd.GeoDataFrame): regions with which features are joined
+            features (gpd.GeoDataFrame): features to be joined
+
+        Returns:
+            JoinerResult object, which contains 3 GeoDataFrames:
+            * original regions
+            * original features
+            * intersection of regions and features, which contains a MultiIndex and a geometry with
+              the intersection
+
+        """
+        joined_parts = []
+
+        for _, single in features.groupby(features["geometry"].geom_type):
```

Comment:
What problem does this solve? Would a single spatial join not work?

Reply:
I could not force an `sjoin` into returning an intersection in a geometry field, but this may be my mistake. And `overlay` is limited to single geometry type in a whole gdf.

Reply:
Oh, ok. I thought we wanted to only do sjoins here, and leave the potential processing (like intersecting geometries) to something further down the pipeline.

Reply:
Maybe I jump ahead with an intersection, then single `sjoin` will do the trick ;)

Reply:
No, I think this is fine. I just needed an explanation

Reply:
> I could not force an `sjoin` into returning an intersection in a geometry field

hmm, that's interesting. That's what sjoin should do imo

Reply:
I would too, but I always get an original geometry from either one of the gdfs